modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -33,13 +35,13 @@ def do_remove(name, type)
       nil
     end
 
-    def execute(cmd, msg = nil, error_only = false)
+    def execute(*args, **keyword_args)
       tsecs = 5
       response = nil
       interpreter = Proxy::SETTINGS.x86_64 ? 'c:\windows\sysnative\cmd.exe' : 'c:\windows\system32\cmd.exe'
-      command = interpreter + ' /c c:\Windows\System32\dnscmd.exe ' + "#{@server} #{cmd}"
-
+      command = interpreter + ' /c c:\Windows\System32\dnscmd.exe ' + "#{@server}" + args.join(" ")
```

Comment:
You no longer pass "/c xxxxx\dnscmd.exe" so remove this. I think you can drop the command variable and just do this below:

    logger.debug "executing: #{interpreter} #{args.join(" ")}"

Or better

    logger.debug { "executing: #{interpreter} #{args.join(" ")}" }